docs/arch/scheduler.md

```diff
@@ -0,0 +1,28 @@
+# Scheduler
+
+## What?
+
+- A scheduler, its job is to schedule a job to an agent. This job is given by a controller
+- A scheduler does NOT create or launch agents.
+- A scheduler can work without any agents.
+- A scheduler knows the current state / capacity (memory, CPU) of each agent always.
+- A scheduler can receive more shares than we have agents
+- A scheduler can distribute task in different agent if it is up or not
+- A scheduler don’t have order for execute task
+
+## Why?
+
+- In order to perform steps and provide the result to the controller
+- To know all the available resources and be able to assign tasks accordingly
+- To manage agents
+- To optimally schedule steps on multiple agents based on agents’ resource capacities and current state.
+- To know which agent is available and which not
+
+## How?
+
+- Agents connect to the scheduler
+- Scheduler saves his actual pool of agents and their state, thos information are no more relevant if the scheduler shutdown
```

Comment:
So if the agent disconnect, you keep its information ? Shouldn't we forget every piece of information about the agent. The agent will reconnect later and ack to the scheduler by itself. 
Or there could be a process to try to reconnect to an agent ?

Reply:
Sorry if it wasn't clear enough, the doc is being updated ;

If an agent is disconnected from a scheduler, then the scheduler will know. The dead agent is removed from the list of registered agents, and the resource pool is updated.

As you say : "we forget every piece of information about the agent"


There is no "process to try to reconnect to an agent", because it is the agent who connects to the scheduler